### ISSUE TITLE
Prometheus configuration incompatibility between collector with TA and NO TA

### ISSUE THREAD

**Issue by murubhas** [open-telemetry/opentelemetry-operator#1622]
Below configuration in `deployment` mode with TA disabled working good. However, when I change the deployment mode to `statefulset` with `TA enabled` it fails with below errors

**TA Error**
```
{"level":"error","ts":1680547146.8650134,"logger":"setup","msg":"Unable to load configuration","error":"error unmarshaling YAML: \"__param_$$1\" is invalid 'replacement' for labelmap action","stacktrace":"main.main\n\t/app/main.go:74\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
{"level":"info","ts":1680547146.8650668,"msg":"Starting the Target Allocator"}
{"level":"info","ts":1680547146.865076,"logger":"allocator","msg":"Unrecognized filter strategy; filtering disabled"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x88 pc=0x24c429f]

goroutine 64 [running]:
github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/target.(*Discoverer).ApplyConfig(0xc0003385a0, 0x0?, 0x0)
	/app/target/discovery.go:75 +0x15f
main.main.func7()
	/app/main.go:151 +0x55
github.com/oklog/run.(*Group).Run.func1({0xc000338960?, 0xc000926180?})
	/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:38 +0x2f
created by github.com/oklog/run.(*Group).Run
	/go/pkg/mod/github.com/oklog/run@v1.1.0/group.go:37 +0x22a
```
**Collector Error**
```
2023/04/03 18:41:58 ADOT Collector version: v0.26.1
2023/04/03 18:41:58 found no extra config, skip it, err: open /opt/aws/aws-otel-collector/etc/extracfg.txt: no such file or directory
Error: invalid configuration: no receiver configuration specified in config
2023/04/03 18:41:58 application run finished with error: invalid configuration: no receiver configuration specified in config
```
**Config Deployment**

```Yaml
---
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: my-metric
  namespace: adot
spec:
  mode: deployment
  serviceAccount: default
  targetAllocator:
    enabled: false
    serviceAccount: default
    filterStrategy: relabel-config
    prometheusCR:
      enabled: false
  env:
    - name: KUBERNETES_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace
    - name: "K8S_NODE_NAME"
      valueFrom:
        fieldRef:
          fieldPath: "spec.nodeName"
  config: |
    receivers:
      prometheus:
        config:
          global:
            scrape_interval: 15s
            scrape_timeout: 10s
          scrape_configs:
          - job_name: kubernetes-service-endpoints
            kubernetes_sd_configs:
            - role: endpoints
            relabel_configs:
            - action: keep
              regex: true
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_scrape
            - action: replace
              regex: (https?)
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_scheme
              target_label: __scheme__
            - action: replace
              regex: (.+)
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_path
              target_label: __metrics_path__
            - action: replace
              regex: ([^:]+)(?::\d+)?;(\d+)
              replacement: $$1:$$2
              source_labels:
              - __address__
              - __meta_kubernetes_service_annotation_prometheus_io_port
              target_label: __address__
            - action: labelmap
              regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
              replacement: __param_$$1
            - action: labelmap
              regex: __meta_kubernetes_service_label_(.+)
            - action: replace
              source_labels:
              - __meta_kubernetes_namespace
              target_label: kubernetes_namespace
            - action: replace
              source_labels:
              - __meta_kubernetes_service_name
              target_label: kubernetes_name
            - action: replace
              source_labels:
              - __meta_kubernetes_pod_node_name
              target_label: kubernetes_node

    processors:
      batch/metrics:
        timeout: 60s  

    exporters:
      logging:
        verbosity: detailed
        sampling_initial: 5
        sampling_thereafter: 15

    service:
      pipelines:
        metrics:
          receivers: [prometheus]
          processors: [batch/metrics]
          exporters: [logging]
      telemetry:
        logs:
          level: debug
```
**Config Statefulset With TA** that is causing error
```yaml
---
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: my-metric
  namespace: adot
spec:
  mode: statefulset
  serviceAccount: default
  targetAllocator:
    enabled: true
    serviceAccount: default
    filterStrategy: relabel-config
    prometheusCR:
      enabled: false
  env:
    - name: KUBERNETES_NAMESPACE
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace
    - name: "K8S_NODE_NAME"
      valueFrom:
        fieldRef:
          fieldPath: "spec.nodeName"
  config: |
    receivers:
      prometheus:
        config:
          global:
            scrape_interval: 15s
            scrape_timeout: 10s
          scrape_configs:
          - job_name: kubernetes-service-endpoints
            kubernetes_sd_configs:
            - role: endpoints
            relabel_configs:
            - action: keep
              regex: true
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_scrape
            - action: replace
              regex: (https?)
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_scheme
              target_label: __scheme__
            - action: replace
              regex: (.+)
              source_labels:
              - __meta_kubernetes_service_annotation_prometheus_io_path
              target_label: __metrics_path__
            - action: replace
              regex: ([^:]+)(?::\d+)?;(\d+)
              replacement: $$1:$$2
              source_labels:
              - __address__
              - __meta_kubernetes_service_annotation_prometheus_io_port
              target_label: __address__
            - action: labelmap
              regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
              replacement: __param_$$1
            - action: labelmap
              regex: __meta_kubernetes_service_label_(.+)
            - action: replace
              source_labels:
              - __meta_kubernetes_namespace
              target_label: kubernetes_namespace
            - action: replace
              source_labels:
              - __meta_kubernetes_service_name
              target_label: kubernetes_name
            - action: replace
              source_labels:
              - __meta_kubernetes_pod_node_name
              target_label: kubernetes_node

    processors:
      batch/metrics:
        timeout: 60s  

    exporters:
      logging:
        verbosity: detailed
        sampling_initial: 5
        sampling_thereafter: 15

    service:
      pipelines:
        metrics:
          receivers: [prometheus]
          processors: [batch/metrics]
          exporters: [logging]
      telemetry:
        logs:
          level: debug
---

```
![image](https://user-images.githubusercontent.com/126094112/229598835-ecdd1635-c530-421b-aa0a-3c0be726fabd.png)

**Comment by matej-g** [comment]
Looks like this duplicates / relates to:
- https://github.com/open-telemetry/opentelemetry-operator/issues/1623
- https://github.com/open-telemetry/opentelemetry-operator/issues/958

**Comment by jaronoff97** [comment]
I think this is resolved by #1712 thank you!
